ui/store/actions.test.js

```diff
@@ -1231,7 +1237,79 @@ describe('Actions', () => {
         },
       ];
 
-      await store.dispatch(actions.setRpcTarget());
+      await store.dispatch(actions.setActiveNetwork());
+      expect(store.getActions()).toStrictEqual(expectedActions);
+    });
+  });
+
+  describe('#editAndSetNetworkConfiguration', () => {
+    afterEach(() => {
+      sinon.restore();
+    });
+
+    it('calls editAndSetNetworkConfiguration', async () => {
+      const store = mockStore();
+
+      const removeNetworkConfigurationStub = sinon
+        .stub()
+        .callsFake((_, cb) => cb());
+
+      const upsertNetworkConfigurationStub = sinon
+        .stub()
+        .callsFake((_, cb) => cb());
+
+      background.getApi.returns({
+        removeNetworkConfiguration: removeNetworkConfigurationStub,
+        upsertNetworkConfiguration: upsertNetworkConfigurationStub,
+      });
+      _setBackgroundConnection(background.getApi());
+
+      await store.dispatch(
+        actions.editAndSetNetworkConfiguration({
+          networkConfigurationId: 'networkConfigurationId',
+          rpcUrl: 'newRpc',
+          chainId: '0x',
+          ticker: 'ETH',
+          nickname: 'nickname',
+          rpcPrefs: { blockExplorerUrl: 'etherscan.io' },
+        }),
+      );
+      expect(removeNetworkConfigurationStub.callCount).toStrictEqual(1);
```

Comment:
Nit: It might be worthwhile confirming that these were called with the expected parameters as well

Reply:
done here: https://github.com/MetaMask/metamask-extension/pull/17421/commits/ccfc91787ab01012269c2d2bd97fa9b3c0de5ace